### PR TITLE
fix: aggregator 2x bump in mainnet

### LINKED ISSUE #1540: hotfix(aggregator): aggregator is bumping to much in mainnet
# Aggregator is bumping to much in mainnet

First, check the parameters being used in mainnet vs testnet. Because in testnet is seems to be working as expected

### PR DESCRIPTION
**Pull request by MarcosNicolau**
### TL;DR
This issue arises because of the mixed use of dynamic transactions and legacy transactions between the simulation and the actual transaction. We fixed it by avoiding the usage of `simulation.GasPrice` as `previousGasPrice` in the first iteration as it returns the `baseFee` * 2 (doubling the gas price).

### Detailed explanation

In the simulation, the `sendResponseTransaction` was executed without a specified `tx.GasPrice()`. This made sense, as simulations do not require actual gas pricing. When no `gasPrice` is provided, the library treats the transaction as dynamic.

In a dynamic transaction, the library sets:
- `priorityFee` = suggestPriorityFee()
- `maxFee` = 2 × BaseFee

This approach is standard practice, as setting `maxFee = 2 × baseFee` ensures the transaction remains valid for approximately six blocks.

On the first iteration, we were setting `PreviousGasPrice` to the value of `simulation.GasPrice()`. However, for dynamic transactions, calling GasPrice() returns the `maxFee`. In subsequent steps, the code calculates the new transaction’s gas price as the maximum of `calculatedGasPrice` and `previousGasPrice`.

Since `previousGasPrice` was erroneously set to `maxFee` (`2 × BaseFee`), the resulting gas price was being doubled.

### Why This Didn’t Appear Earlier

The issue went unnoticed in Holesky due to its extremely low base gas cost—four orders of magnitude smaller than typical gas prices. Under such conditions, the 2 × BaseFee did not significantly interfere with calculatedGasPrice.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix

## Checklist

- [ ] “Hotfix” to `testnet`, everything else to `staging`
- [ ] Linked to Github Issue
- [ ] This change depends on code or research by an external entity
  - [ ] Acknowledgements were updated to give credit
- [ ] Unit tests added
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] This change is an Optimization
  - [ ] Benchmarks added/run
- [ ] Has a known issue
  - [Link to the open issue addressing it]() 
- [ ] If your PR changes the Operator compatibility (Ex: Upgrade prover versions)
  - [ ] This PR adds compatibility for operator for both versions and do not change batcher/docs/examples
  - [ ] This PR updates batcher and docs/examples to the newer version. This requires the operator are already updated to be compatible

### CODE REVIEW

File: core/chainio/avs_writer.go [hunk 1]
```diff
@@ -113,7 +113,16 @@ func (w *AvsWriter) SendAggregatedResponse(batchIdentifierHash [32]byte, batchMe
...
 		}
-		previousTxGasPrice := txOpts.GasPrice
+
+		// if txOpts.GasPrice wasn't previously set use the fetched gasPrice
+		// this should happen on the first iteration only
+		var previousTxGasPrice *big.Int
+		if txOpts.GasPrice == nil {
+			previousTxGasPrice = gasPrice
+		} else {
+			previousTxGasPrice = gasPrice
```
Comment (Oppen) [inline core/chainio/avs_writer.go]: I believe in this case you meant `txOpts.GasPrice`?

Comment (MarcosNicolau) [reply core/chainio/avs_writer.go]: Oh yes thanks!

**Review by Oppen — APPROVED** [review]
LGTM

**Review by uri-99 — APPROVED** [review]

**Review by JulianVentura — APPROVED** [review]
LGTM

**Review by JuArce — APPROVED** [review]
